### PR TITLE
refactor(async): align the error messages to the style guide

### PR DESCRIPTION
**Pull request by irbull**
Aligns the error messages in the `async` folder to match the style guide.

https://github.com/denoland/std/issues/5574

### CODE REVIEW

File: async/retry.ts [hunk 1]
```diff
@@ -134,11 +134,21 @@ export async function retry<T>(
     ...opts,
   };
 
-  if (options.maxTimeout <= 0) throw new TypeError("maxTimeout is less than 0");
+  if (options.maxTimeout <= 0) {
+    throw new TypeError(
+      `'maxTimeout' must be positive: current value is ${options.maxTimeout}`,
```
Comment (kt3k) [inline async/retry.ts:139]: I would add the below based on the rule 4:

> 4. Message should state the action that lead to the error:

```suggestion
      `Cannot retry as 'maxTimeout' must be positive: current value is ${options.maxTimeout}`,
```

File: async/retry.ts [hunk 2]
```diff
@@ -134,11 +134,21 @@ export async function retry<T>(
...
+    );
+  }
   if (options.minTimeout > options.maxTimeout) {
-    throw new TypeError("minTimeout is greater than maxTimeout");
+    throw new TypeError(
+      `'minTimeout' must be <= 'maxTimeout': current values 'minTimeout=${options.minTimeout}', 'maxTimeout=${options.maxTimeout}'`,
```
Comment (kt3k) [inline async/retry.ts:144]:
```suggestion
      `Cannot retry as 'minTimeout' must be <= 'maxTimeout': current values 'minTimeout=${options.minTimeout}', 'maxTimeout=${options.maxTimeout}'`,
```

File: async/retry.ts [hunk 3]
```diff
@@ -134,11 +134,21 @@ export async function retry<T>(
...
+    );
+  }
+  if (options.jitter > 1) {
+    throw new TypeError(
+      `'jitter' must be <= 1: current value is ${options.jitter}`,
```
Comment (kt3k) [inline async/retry.ts:149]:
```suggestion
      `Cannot retry as 'jitter' must be <= 1: current value is ${options.jitter}`,
```

File: async/pool.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@
 // This module is browser compatible.
 
 /** Error message emitted from the thrown error while mapping. */
-const ERROR_WHILE_MAPPING_MESSAGE = "Threw while mapping.";
+const ERROR_WHILE_MAPPING_MESSAGE = "Threw while mapping";
```
Comment (kt3k) [inline async/pool.ts:5]:
```suggestion
const ERROR_WHILE_MAPPING_MESSAGE = "Cannot complete the mapping as an error was thrown from an item";
```

File: async/retry_test.ts [hunk 5]
```diff
@@ -92,7 +92,7 @@ Deno.test(
           jitter: 2,
         }),
       TypeError,
-      "jitter is greater than 1",
+      "'jitter' must be <= 1: current value is 2",
```
Comment (irbull) [inline async/retry_test.ts:95]:
```suggestion
      "Cannot retry as 'jitter' must be <= 1: current value is 2",
```

File: async/retry_test.ts [hunk 6]
```diff
@@ -78,7 +78,7 @@ Deno.test(
           maxTimeout: -1,
         }),
       TypeError,
-      "maxTimeout is less than 0",
+      "'maxTimeout' must be positive: current value is -1",
```
Comment (irbull) [inline async/retry_test.ts:81]:
```suggestion
      "Cannot retry as 'maxTimeout' must be positive: current value is -1",
```

File: async/pool_test.ts [hunk 7]
```diff
@@ -38,7 +38,7 @@ Deno.test("pooledMap() handles errors", async () => {
       }
     },
     AggregateError,
-    "Threw while mapping.",
+    "Threw while mapping",
```
Comment (irbull) [inline async/pool_test.ts:41]:
```suggestion
    "Cannot complete the mapping as an error was thrown from an item",
```

File: async/retry_test.ts [hunk 8]
```diff
@@ -64,7 +64,7 @@ Deno.test(
           maxTimeout: 100,
         }),
       TypeError,
-      "minTimeout is greater than maxTimeout",
+      "'minTimeout' must be <= 'maxTimeout': current values 'minTimeout=1000', 'maxTimeout=100'",
```
Comment (irbull) [inline async/retry_test.ts:67]:
```suggestion
      "Cannot retry as 'minTimeout' must be <= 'maxTimeout': current values 'minTimeout=1000', 'maxTimeout=100'",
```

**Review by kt3k — APPROVED** [review]
LGTM. Thanks for updating!

**Pull request merged**